server/command.go

```diff
@@ -121,6 +125,90 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*mo
 		}
 		p.postCommandResponse(args, str.String())
 		return &model.CommandResponse{}, nil
+	case "set_channel_welcome":
+		if len(parameters) == 0 {
+			p.postCommandResponse(args, "`set_channel_welcome` command requires the message to be provided")
+			return &model.CommandResponse{}, nil
+		}
+
+		if channelInfo, appErr := p.API.GetChannel(args.ChannelId); appErr != nil {
+			errMsg := fmt.Sprintf(
+				"error occured while checking the type of the chanelId `%s`: `%s`",
+				args.ChannelId, err)
+			p.postCommandResponse(args, errMsg)
+			return &model.CommandResponse{}, nil
+		} else if channelInfo.Type == model.CHANNEL_PRIVATE {
+			p.postCommandResponse(
+				args, "welcome messages are not supported for direct channels")
+			return &model.CommandResponse{}, nil
+		}
+
+		// strings.Fields will consume ALL whitespace, so plain re-joining of the
+		// parameters slice will not produce the same message
+		message := strings.SplitN(args.Command, "set_channel_welcome", 2)[1]
```

Comment:
Would a simple `string.TrimPrefix(args.Command, "set_channel_welcome")` be enough here?

Reply:
No, it would not. The `args.Command` contains both the `/welcomebot` part and the `set_channel_welcome` part and we do not know how many spaces there are between these two. See [1] for an example.

[1] https://play.golang.org/p/OA5NifHz_vf